README.md

```diff
@@ -286,52 +292,169 @@ type Action = {
+
+Verifications must be opted in to by both the external identity (i.e. Ethereum address) and the Farcaster account. The bi-directional verification is implemented in two steps:
+
+1. Create a `VerificationClaim` object that contains the external address and the Farcaster account. Have the external identity sign a hash of the claim object, creating a directional proof from the external identity to the Farcaster account.
+2. Create a `VerificationAdd` message containing the external address and signature from step 1. Have the Farcaster account sign the message and share it with the network, wrapping and verifying the verification claim.
+
+In order to validate the `VerificationAdd` message, Farcaster hubs evaluate both the Farcaster signature as well as the external identity's signature. Initially, external signatures must implement [EIP 191 version 0x45](https://eips.ethereum.org/EIPS/eip-191) (i.e. Ethereum's `personal_sign`).
+
+Here are the schemas for `VerificationClaim` and `VerificationAdd`:
+
+```ts
+type VerificationClaim = {
+  externalAddressUri: string;
+  account: number;
+};
+
+type VerificationAdd = {
+  message: {
+    body: {
+      externalAddressUri: string;
+      claimHash: string;
+      externalSignature: string;
+      externalSignatureType: "eip-191-0x45";
+      schema: "farcaster.xyz/schemas/v1/verification-add";
+    };
+    account: number;
+    timestamp: number;
+  };
+  envelope: {
+    hash: string;
+    hashType: "BLAKE2b";
+    signature: string;
+    signatureType: "ed25519" | "ecdsa-secp256k1";
+    signerPubKey: string;
+  };
+};
+```
+
+Verifications are stored in the Verifications Set, which is a modified [Last-Write-Wins-Element Set](<https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type#LWW-Element-Set_(Last-Write-Wins-Element-Set)>). Valid and unremoved `VerificationAdd` messages are stored in the `adds` set, indexed by their `claimHash`, a hash of the `VerificationClaim`. Any Farcaster account and external identity pair will have the same `claimHash`.
+
+Verifications can be removed by sending a valid `VerificationRemove` message to the network that references the `claimHash` to remove.
+
+```ts
+type VerificationRemove = {
+  message: {
+    body: {
+      claimHash: string;
+      schema: "farcaster.xyz/schemas/v1/verification-remove";
+    };
+    account: number;
+    timestamp: number;
+  };
+  envelope: {
+    hash: string;
+    hashType: "BLAKE2b";
+    signature: string;
+    signatureType: "ed25519" | "ecdsa-secp256k1";
+    signerPubKey: string;
+  };
+};
+```
+
+A valid `VerificationRemove` message is stored in the `removes` set, indexed by `claimHash`, and removes `claimHash` and the relevant `VerificationAdd` message from the `adds` set. A `claimHash` cannot be in both the `adds` set and the `removes` set, and conflicts are resolved using message timestamps.
```

Comment:
This section is a bit difficult to read given all the inline code blocks, and also overlaps a bit with the pseudocode section. 

I wonder if we can simplify it to just describe the high level ideas here which would be: 

1. A user can add a verification at any time by broadcasting ____ 
2. They can remove and re-add it at any time by broadcasting ____ 
3. When removed the data is redacted/ effectively hiddne because _____ 

and then we can get into the nuances in the code section.

Reply:
I re-wrote this section, and I think it's easier to understand now